api/clients/eigenda_client.go

```diff
@@ -84,27 +93,30 @@ func (m EigenDAClient) GetCodec() codecs.BlobCodec {
 // data, which is necessary for generating KZG proofs for data's correctness.
 // The function handles potential errors during blob retrieval, data length
 // checks, and decoding processes.
+// TODO: should we use a pointer receiver instead, to prevent unnecessary copying of the EigenDAClient struct?
 func (m EigenDAClient) GetBlob(ctx context.Context, batchHeaderHash []byte, blobIndex uint32) ([]byte, error) {
 	data, err := m.Client.RetrieveBlob(ctx, batchHeaderHash, blobIndex)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not retrieve blob: %w", err)
 	}
 
 	if len(data) == 0 {
+		// TODO: explain when/why/how this can happen
 		return nil, fmt.Errorf("blob has length zero")
 	}
```

Comment:
@bxue-l2 do you know?

Reply:
rip, @jianoaix @ian-shim any ideas here?

Reply:
Empty blobs should have been rejected from entering the system: https://github.com/Layr-Labs/eigenda/blob/master/disperser/apiserver/server.go#L930

Reply:
Thanks: updated [5bb9efc](https://github.com/Layr-Labs/eigenda/pull/780/commits/5bb9efc3b69dfde838eb1a5bcbed41b34f191aa1)